practice/pull-request-review-practice.md

```diff
@@ -0,0 +1,83 @@
+
+## Lesson Plan
+
+### Activity 1: Pull Request Basics
+**Task:** Each participant will create a pull request for a basic change in a shared repository.  
+
+1. **Instructions:**
+   - Add your name to the [participants.md file](https://github.com/Techtonica/curriculum/blob/main/practice/participants.md) in the Techtonica curriculum repository.
+   - Submit a pull request with:
+     - **Title:** A concise, descriptive title.
+     - **Description:** A clear explanation of the change.
+     - **Visuals:** Screenshots with before-changes and after-changes labels in the description.
+
+2. **Outcome:** 
+   - Participants practice creating detailed pull requests.
+
+---
+
+### Activity 2: Interactive PR Review Process
+**Task:** Participants will engage in a two-way PR review process using their Week 7 full stack application code.  
+
+#### Repo Owner:
+1. **Push a commit:** Remove all comments from your Week 7 application.
+2. **Review:** Evaluate incoming changes from your feedback partner’s pull request, leaving comments as needed.
+
+#### Collaborator:
+1. **Add comments:** Create a branch and add comments explaining what each line of code does in the repo owner’s project.
+2. **Submit a pull request:** Submit these changes to the repo owner for review.
+
+---
+
+### Activity 3: Addressing PR Feedback
+**Task:** Pull request owners must address feedback using one of the following methods:  
+- **Acknowledged:** React to the feedback with an emoji to show it is accepted.  
+- **Replied:** Respond to comments by explaining your approach or adding items to a backlog for future review.  
+- **Resolved:** Make updates directly in the PR and link the new commit to the comment, marking the conversation as resolved.  
+
+---
+
+## Questions to Consider
+- How does the PR review process enhance code quality and collaboration?  
+- How can you break down work and share progress effectively with your trio or mentor?  
+
+## Common Mistakes / Misconceptions
+- **Overlooking the PR structure:** Avoid unclear titles or missing descriptions in PRs.  
+- **Ignoring feedback:** Actively engage with feedback to show accountability.  
+- **Miscommunication in collaboration:** Ensure clarity when describing changes or feedback.  
+
+## Supplemental Materials
+- [GitHub Documentation: Creating a Pull Request](https://docs.github.com/en/pull-requests)  
+- [Best Practices for Code Review](https://google.github.io/eng-practices/review/)  
+
+## Real-World Application
+The ability to read, understand, and critique others’ code is a critical skill for software development. This exercise mirrors professional workflows where collaboration and peer review ensure the quality and maintainability of software projects.  
+
+## Independent Practice
+Participants will continue working on their Week 8 project (Game with API) while practicing the PR review process to balance individual and collaborative work effectively.
+
+## Challenge
```

Comment:
It could be beneficial to move the real-world application, challenge, and mistakes up into the lesson ahead of giving tasks.

Reply:
@daaimah123 done [here](https://github.com/Techtonica/curriculum/pull/2145/commits/0f7fde712d4087429e7919392a18237a653f5d48).